grpc.go

```diff
@@ -1255,6 +1266,33 @@ func (a *agentGRPC) CreateSandbox(ctx context.Context, req *pb.CreateSandboxRequ
 	return emptyResp, nil
 }
 
+// flushAll flushes all system journal data.
+func flushAll() error {
+	args := []string{"journalctl", "--flush", "--sync"}
+
+	cmd := exec.Command(args[0], args[1:]...)
+
+	return cmd.Run()
+}
+
+// requestServerStop asks the goroutine that started the server goroutine
+// to end the gRPC server.
+func (a *agentGRPC) requestServerStop() {
+	// Synchronize the caches on the system. This is needed to ensure
+	// there is no pending transactions left before the VM is shut down.
+	syscall.Sync()
+
+	if tracing {
+		err := flushAll()
+		if err != nil {
+			agentLog.WithError(err).Warn("failed to flush")
+		}
+	}
+
+	// Inform the caller the server should end
+	a.sandbox.shutdown <- true
```

Comment:
Not using a boolean Go channel, but a struct{} instead would be more appropriate. This would lead to modify this line using:
```go
close(a.sandbox.shutdown)
```